Fix undefined behavior due to modifying PL_na twice between sequence points.
(This might also make the code just a tiny bit faster.) Parenthesize macro
parameters.

Index: src/perl/perl-signals.c
===================================================================
--- src/perl/perl-signals.c	(revision 4958)
+++ src/perl/perl-signals.c	(working copy)
@@ -433,9 +433,9 @@
         perl_signal_destroy(rec);
 }
 
-#define sv_func_cmp(f1, f2, len) \
-	(f1 == f2 || (SvPOK(f1) && SvPOK(f2) && \
-		strcmp((char *) SvPV(f1, len), (char *) SvPV(f2, len)) == 0))
+#define sv_func_cmp(f1, f2) \
+	((f1) == (f2) || (SvPOK(f1) && SvPOK(f2) && \
+		strcmp((char *) SvPV_nolen(f1), (char *) SvPV_nolen(f2)) == 0))
 
 static void perl_signal_remove_list(GSList **list, SV *func)
 {
@@ -444,7 +444,7 @@
 	for (tmp = *list; tmp != NULL; tmp = tmp->next) {
 		PERL_SIGNAL_REC *rec = tmp->data;
 
-		if (sv_func_cmp(rec->func, func, PL_na)) {
+		if (sv_func_cmp(rec->func, func)) {
 			perl_signal_remove_list_one(list, rec);
 			break;
 		}
